WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java

```diff
@@ -161,27 +114,69 @@ public void onClick(View v) {
             }
         });
 
+        getSupportFragmentManager().addOnBackStackChangedListener(new FragmentManager.OnBackStackChangedListener() {
+            @Override
+            public void onBackStackChanged() {
+                if (getSupportFragmentManager().getBackStackEntryCount() == 0) {
+                    mViewModel.setTitle(getString(R.string.plugins));
+                }
+            }
+        });
+
         configureRecycler(mSitePluginsRecycler);
         configureRecycler(mPopularPluginsRecycler);
         configureRecycler(mNewPluginsRecycler);
 
-        List<SitePluginModel> sitePlugins = mPluginStore.getSitePlugins(mSite);
-        for (SitePluginModel plugin: sitePlugins) {
-            mSitePluginsMap.put(plugin.getSlug(), plugin);
-        }
-        reloadAllPlugins();
+        mViewModel.start();
+        setupObservers();
+    }
 
-        if (savedInstanceState == null) {
-            fetchPlugins(PluginListType.SITE, false);
-            fetchPlugins(PluginListType.POPULAR, false);
-            fetchPlugins(PluginListType.NEW, false);
-        }
+    private void setupObservers() {
+        mViewModel.getTitle().observe(this, new Observer<String>() {
+            @Override
+            public void onChanged(@Nullable String title) {
+                setTitle(title);
+            }
+        });
 
-        getFragmentManager().addOnBackStackChangedListener(new FragmentManager.OnBackStackChangedListener() {
+        mViewModel.getSitePlugins().observe(this, new Observer<List<SitePluginModel>>() {
             @Override
-            public void onBackStackChanged() {
-                if (getFragmentManager().getBackStackEntryCount() == 0) {
-                    setTitle(R.string.plugins);
+            public void onChanged(@Nullable final List<SitePluginModel> sitePlugins) {
+                refreshPluginAdapterAndVisibility(PluginListType.SITE, sitePlugins);
+            }
+        });
+
+        mViewModel.getNewPlugins().observe(this, new Observer<List<WPOrgPluginModel>>() {
+            @Override
+            public void onChanged(@Nullable final List<WPOrgPluginModel> newPlugins) {
+                refreshPluginAdapterAndVisibility(PluginListType.NEW, newPlugins);
+            }
+        });
+
+        mViewModel.getPopularPlugins().observe(this, new Observer<List<WPOrgPluginModel>>() {
+            @Override
+            public void onChanged(@Nullable final List<WPOrgPluginModel> popularPlugins) {
+                refreshPluginAdapterAndVisibility(PluginListType.POPULAR, popularPlugins);
+            }
+        });
+
+        mViewModel.getSitePluginsListStatus().observe(this, new Observer<PluginBrowserViewModel.PluginListStatus>() {
+            @Override
+            public void onChanged(@Nullable PluginBrowserViewModel.PluginListStatus listStatus) {
+                showProgress(listStatus == PluginBrowserViewModel.PluginListStatus.FETCHING
```

Comment:
This also creates a synthetic accessor method.  We can make the `showProgress` method `protected` or package-private to avoid that.

Reply:
All synthetic accessor method issues should be fixed in 8244bd59d148e7e560c82951daded148e249208f.